test/JDBC/expected/1_grant_schema.out

```diff
@@ -0,0 +1,2936 @@
+go
+~~START~~
+int
+1
+~~END~~
+
+exec s1.p1;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from s1.f1()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql user=l2 password=123
+select current_user;
+go
+~~START~~
+varchar
+u2
+~~END~~
+
+select * from s1.t1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+insert into s1.t1 (a) values (next value for s1.sq1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+delete from s1.t1 where a = 1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t1)~~
+
+select * from s1.v1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v1)~~
+
+exec s1.p1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p1)~~
+
+select * from s1.f1()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f1)~~
+
+
+-- tsql
+-- create new objects in the schema s1
+create table s1.t2(a int);
+go
+create view s1.v2 as select 1;
+go
+create procedure s1.p2 as select 1;
+go
+create function s1.f2() returns int begin declare @a int; set @a = 1; return @a; end
+go
+
+-- tsql user=l1 password=123
+-- Test Group 5[user u1 has select and execute privilege on the schema]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t2;
+go
+~~START~~
+int
+~~END~~
+
+insert into s1.t2 values (1);
+go
+~~ROW COUNT: 1~~
+
+delete from s1.t2 where a = 1;
+go
+~~ROW COUNT: 1~~
+
+select * from s1.v2;
+go
+~~START~~
+int
+1
+~~END~~
+
+exec s1.p2;
+go
+~~START~~
+int
+1
+~~END~~
+
+select * from s1.f2()
+go
+~~START~~
+int
+1
+~~END~~
+
+
+-- tsql user=l2 password=123
+select current_user;
+go
+~~START~~
+varchar
+u2
+~~END~~
+
+select * from s1.t2;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t2)~~
+
+insert into s1.t2 values (1);
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t2)~~
+
+delete from s1.t2 where a = 1;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for table t2)~~
+
+select * from s1.v2;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for view v2)~~
+
+exec s1.p2;
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for procedure p2)~~
+
+select * from s1.f2()
+go
+~~ERROR (Code: 33557097)~~
+
+~~ERROR (Message: permission denied for function f2)~~
+
+
+-- tsql
+revoke select on schema::s1 from u1;
+go
+revoke insert on schema::s1 from u1;
+go
+revoke delete on schema::s1 from u1;
+go
+revoke update on schema::s1 from u1;
+go
+
+-- tsql user=l1 password=123
+-- Test Group 6 
+-- [user u1 has execute privilege on the schema, u1 has select privilege on s1.t1 via trigger]
+select current_user;
+go
+~~START~~
+varchar
+u1
+~~END~~
+
+select * from s1.t1;
```

Comment:
This is expected because trigger tr1 was triggered when we did insert previously in s1.t1 which explicitly granted u1 select permission to s1.t1, right?

Reply:
Yes. This was added to show how GRANT would work within a trigger.